src/languages/fortran.js

```diff
@@ -71,9 +71,9 @@ function(hljs) {
       },
       {
         className: 'number',
-        begin: '-?(\\d+(\\.\\d*)?|\\.\\d+)([DdEe][+-]?\\d+)?',
+        begin: '(?=\\b|\\+|\\-|\\.)(?=\\.\\d|\\d)(?:\\d+)?(?:\\.?\\d*)(?:[de][+-]?\\d+)?\\b\\.?',
```

Comment:
If i understand correctly your question, my answer is: Yes, We need them. Because `.\d+` (like .23) can,  sadly, be a number as `\d+.` (12.).
If we only take the digit, it will be strange at first look, no? The number is `.23` not `23` .  I guess is a personal preference.

It's me to thank you for your amazing work!